### PR TITLE
feat: jump server implementation

### PR DESCRIPTION
**Pull request by jacobdotcosta**

### CODE REVIEW

File: openshift/ocp_on_openstack.adoc [hunk 1]
```diff
@@ -89,6 +89,83 @@ Push the `pass` changes to the git repository.
...
+=== Install jump server
+
+[source,bash]
+----
+ansible-playbook ansible/playbook/openstack/openstack_vm_create_passwordstore.yml \
+  -e '{"openstack": {"vm": {"network": "ocp-sscpc-openshift","image": "Fedora-Cloud-Base-29", "flavor": "m1.small"}}}' \
+  -e vm_name=ocp-jump-server
+----
+
+Create a floating ip from the external network that your router is connected to, in our case `provider_net_cci_13`.
```
Comment (cmoulliard) [inline openshift/ocp_on_openstack.adoc:106]: This point should be clarified as I'm lost between the role of "ocp-sscpc-openshift" and "provider_net_cci_13" !

File: openshift/ocp_on_openstack.adoc [hunk 2]
```diff
@@ -89,6 +89,83 @@ Push the `pass` changes to the git repository.
...
+====
+
+[source,bash]
+----
+openstack server add floating ip <uuid_of_the_create_server> <floating_ip_that_you_got_from_previous_command>
+----
+
+[WARNING]
+====
+Manually add an entry to the passwordstore under `openstack/ocp-jumpserver/floating_ip` with the floating IP.
```
Comment (cmoulliard) [inline openshift/ocp_on_openstack.adoc:136]: As the floating IP will change (and this is perhaps the issue that we have today), we must take care about how to create such floating IP and renew it maybe ?

File: openshift/ocp_on_openstack.adoc [hunk 3]
```diff
@@ -89,6 +89,83 @@ Push the `pass` changes to the git repository.
...
+== Post installation steps
+
+To be able to SSH to any of the OCP cluster machines a _jump server_ is required.
+
+=== Install jump server
+
+[source,bash]
+----
+ansible-playbook ansible/playbook/openstack/openstack_vm_create_passwordstore.yml \
+  -e '{"openstack": {"vm": {"network": "ocp-sscpc-openshift","image": "Fedora-Cloud-Base-29", "flavor": "m1.small"}}}' \
```
[the remark this answers is not in the document]
Comment (jacobdotcosta) [reply openshift/ocp_on_openstack.adoc:102]: The installation process generated a network named `ocp-<code>-openshift` that allows VM instance communication. This is why the _jump server_ must also be generated on this network.

I'm updating the documentation to improve the information on this.

File: openshift/ocp_on_openstack.adoc (hunk 1 above)
Comment (jacobdotcosta) [reply openshift/ocp_on_openstack.adoc:106]: Improving the docs.

File: openshift/ocp_on_openstack.adoc [hunk 4]
```diff
@@ -89,6 +89,123 @@ Push the `pass` changes to the git repository.
...
+== Post installation steps
+
+
+=== Install jump server
+
+To be able to SSH to any of the OCP cluster machines a _jump server_ is required.
+
+==== Requirements
+
+1. Obtain the name of the network interface created by the OCP installation process
```
Comment (cmoulliard) [inline openshift/ocp_on_openstack.adoc:102]: "by the OCP installation process". Is it an ansible process or bash script or ... ? You should be more precise here and ideally point to the documentation about we install ocp

File: openshift/ocp_on_openstack.adoc [hunk 5]
```diff
@@ -89,6 +89,123 @@ Push the `pass` changes to the git repository.
...
+[NOTE]
+====
+For the sake of example the `ocp-sscpc` prefix will be used.
+====
+
+==== Create and configure the server
+
+[WARNING]
+====
+One _jump server_ will be required so the OCP cluster name prefix should be used when creating this VM.
```
Comment (cmoulliard) [inline openshift/ocp_on_openstack.adoc:115]: This verbiage is not clear. Are you saying that we need one jump_server / ocp network ?

File: openshift/ocp_on_openstack.adoc [hunk 6]
```diff
@@ -89,6 +89,123 @@ Push the `pass` changes to the git repository.
...
+====
+
+[source,bash]
+----
+ansible-playbook ansible/playbook/openstack/openstack_vm_create_passwordstore.yml \
+  -e '{"openstack": {"vm": {"network": "ocp-sscpc-openshift","image": "Fedora-Cloud-Base-29", "flavor": "m1.small"}}}' \
+  -e vm_name=ocp-sscpc-jump-server
+----
+
+Create a _Floating IP_ from the external network that one's router is connected to, in our case `provider_net_cci_13`. As the _jump server_ is created on the same network as the OCP cluster it isn't accesible from the outside. To access this server a RHOS floating IP is required. The `provider_net_cci_13` is the team's public RHOS network name.
```
Comment (cmoulliard) [inline openshift/ocp_on_openstack.adoc:125]: Can you explain how you get such an external network ? what is the router you refers here ?

File: openshift/ocp_on_openstack.adoc (hunk 5 above)
Comment (jacobdotcosta) [reply openshift/ocp_on_openstack.adoc:115]: Yes, one jump server for each OCP cluster, as each cluster installation generates it's own network.

File: openshift/ocp_on_openstack.adoc (hunk 4 above)
Comment (jacobdotcosta) [reply openshift/ocp_on_openstack.adoc:102]: This document describes the OCP installation process in it's whole. The installation process is a previous point of the document. I'll try improving the description anyway.

File: openshift/ocp_on_openstack.adoc (hunk 6 above)
Comment (jacobdotcosta) [reply openshift/ocp_on_openstack.adoc]: Improved the docs to better explain this paragraph.

File: openshift/ocp_on_openstack.adoc (hunk 5 above)
Comment (jacobdotcosta) [reply openshift/ocp_on_openstack.adoc]: Improved the docs to explain this.

File: openshift/ocp_on_openstack.adoc [hunk 7]
```diff
@@ -89,6 +96,173 @@ Push the `pass` changes to the git repository.
...
+ansible-playbook ansible/playbook/openstack/openstack_vm_create_passwordstore.yml \
+  -e '{"openstack": {"vm": {"network": "ocp-sscpc-openshift","image": "Fedora-Cloud-Base-29", "flavor": "m1.small"}}}' \
+  -e vm_name=ocp-sscpc-jump-server
+----
+
+Once the host is provisioned create a _Floating IP_ from the external 
+ network that one's router is connected to. 
+ 
+To collect the name of the 
+ _External Network_ go to the `Network > Routers` option of the RHOS console
```
Comment (cmoulliard) [inline openshift/ocp_on_openstack.adoc:168]: The router that we are refering has nothing to do with OCP. This is an Openstack Network Router - https://docs.openstack.org/python-openstackclient/pike/cli/command-objects/router.html. So, I propose to change the wording to mention that we must select using the console (or client maybe ?) the `Network > Routers` corresponding to the network which has been created for the OCP clusters.

Comment (jacobdotcosta) [reply openshift/ocp_on_openstack.adoc]: Improved the wording, hope it's clearer now.

File: openshift/ocp_on_openstack.adoc [hunk 8]
```diff
@@ -89,6 +96,176 @@ Push the `pass` changes to the git repository.
 git push
 ----
 
+== Post installation steps
+
+=== Install jump server
+
+The OCP cluster is created on a private network that allows communication
```
Comment (cmoulliard) [inline openshift/ocp_on_openstack.adoc:103]: "The OCP cluster is created on a private network ..." --> "An OCP cluster, when created on RHOS", is deployed on a private OpenStack network ...

File: openshift/ocp_on_openstack.adoc [hunk 9]
```diff
@@ -89,6 +96,176 @@ Push the `pass` changes to the git repository.
...
+The OCP cluster is created on a private network that allows communication 
+ between the different hosts of the cluster. To be able to SSH to any of 
+ the hosts of this private network a _jump server_ host is required. 
+ This host must be created inside the network of the OCP cluster.
+ 
+To access the _jump server_ a Public IP is required which can be 
+ accomplished by creating a RHOS _Floating IP_ and associating it with that 
+ host.
+
+The OCP private network is created automatically during the installation
```
Comment (cmoulliard) [inline openshift/ocp_on_openstack.adoc:112]: "The OCP private network is created automatically during the installation 
 process." --> "The private network is created automatically during the execution of the OCP playbook creating the cluster on RHOS."

File: openshift/ocp_on_openstack.adoc [hunk 10]
```diff
@@ -89,6 +96,176 @@ Push the `pass` changes to the git repository.
...
+ _External network_ can be identified from the RHOS _Router_ genereated 
+ by the RHOS OCP installation.
+ 
+To collect the name of the _External Network_ go to the
+ `Network > Routers` option of the RHOS console. 
+ The RHOS router for the OCP cluster can be identified by it's name 
+ that should be `<clusterName>-<code>-external-router`.
+ Once the router is identified check the `External Network` column, which in this example would be `ocp-sscpc-external-router`.
+
+The `ocp_cluster` Ansible Role as this value predefined in the installation
```
Comment (cmoulliard) [inline openshift/ocp_on_openstack.adoc:176]: This text is not clear "The `ocp_cluster` Ansible Role as this value predefined in the installation 
procedure and it's `provider_net_cci_13`"

File: openshift/ocp_on_openstack.adoc (hunk 8 above)
Comment (jacobdotcosta) [reply openshift/ocp_on_openstack.adoc]: changed

File: openshift/ocp_on_openstack.adoc (hunk 9 above)
Comment (jacobdotcosta) [reply openshift/ocp_on_openstack.adoc]: changed

File: openshift/ocp_on_openstack.adoc (hunk 10 above)
Comment (jacobdotcosta) [reply openshift/ocp_on_openstack.adoc]: Improved the text to try making it clearer.

**Review by cmoulliard — APPROVED** [review]

**Pull request merged**